packages/starknet-snap/src/state/network-state-manager.test.ts

```diff
@@ -0,0 +1,243 @@
+
+      expect(result).toStrictEqual([STARKNET_SEPOLIA_TESTNET_NETWORK]);
+    });
+
+    it('returns empty array if the account address can not be found', async () => {
+      const chainId = constants.StarknetChainId.SN_SEPOLIA;
+      await mockState({
+        networks: [STARKNET_MAINNET_NETWORK],
+      });
+
+      const stateManager = new NetworkStateManager();
+      const result = await stateManager.list([new ChainIdFilter([chainId])]);
+
+      expect(result).toStrictEqual([]);
+    });
+  });
+
+  describe('updateNetwork', () => {
+    it('updates the network', async () => {
+      const { state } = await mockState({
+        networks: [STARKNET_MAINNET_NETWORK],
+      });
+
+      const stateManager = new NetworkStateManager();
+      const updatedEntity = {
+        ...STARKNET_MAINNET_NETWORK,
+        name: 'Updated Network',
+        nodeUrl: 'http://localhost:8080',
+      };
+      await stateManager.updateNetwork(updatedEntity);
+
+      expect(state.networks[0]).toStrictEqual(updatedEntity);
+    });
+
+    it('throws `Network does not exist` error if the update entity can not be found', async () => {
+      await mockState({
+        networks: [STARKNET_MAINNET_NETWORK],
+      });
+
+      const stateManager = new NetworkStateManager();
+      const updatedEntity = {
+        ...STARKNET_SEPOLIA_TESTNET_NETWORK,
+        name: 'Updated Network',
+        nodeUrl: 'http://localhost:8080',
+      };
+
+      await expect(stateManager.updateNetwork(updatedEntity)).rejects.toThrow(
+        'Network does not exist',
+      );
+    });
+  });
+
+  describe('getCurrentNetwork', () => {
+    it('get the current network', async () => {
+      await mockState({
+        networks: [STARKNET_MAINNET_NETWORK, STARKNET_SEPOLIA_TESTNET_NETWORK],
+        currentNetwork: STARKNET_MAINNET_NETWORK,
+      });
+
+      const stateManager = new NetworkStateManager();
+      const result = await stateManager.getCurrentNetwork();
+
+      expect(result).toStrictEqual(STARKNET_MAINNET_NETWORK);
+    });
+
+    it('returns null if the current network is null or undefined', async () => {
+      await mockState({
+        networks: [STARKNET_MAINNET_NETWORK, STARKNET_SEPOLIA_TESTNET_NETWORK],
+      });
+
+      const stateManager = new NetworkStateManager();
+      const result = await stateManager.getCurrentNetwork();
+
+      expect(result).toBeNull();
+    });
+  });
+
+  describe('setCurrentNetwork', () => {
+    it.each([
+      {
+        status: 'undefined',
+        currentNetwork: undefined,
+        updateTo: STARKNET_SEPOLIA_TESTNET_NETWORK,
+      },
+      {
+        status: 'same',
+        currentNetwork: STARKNET_MAINNET_NETWORK,
+        updateTo: STARKNET_MAINNET_NETWORK,
+      },
+      {
+        status: 'different',
+        currentNetwork: STARKNET_MAINNET_NETWORK,
+        updateTo: STARKNET_SEPOLIA_TESTNET_NETWORK,
+      },
+    ])(
+      'set the current network when the current network is $status',
```

Comment:
what is the use case of `status` variable, I don't see it being used ?

Reply:
it is a generic var replace to this test, if you run the test, it will become "when the current network is same/different/undefined"